src/services/db/GitFileSystemService.ts

```diff
@@ -1236,6 +1241,131 @@ export default class GitFileSystemService {
       })
   }
 
+  // Delete multiple files
+  deleteMultipleFiles(
+    repoName: string,
+    items: Array<{ filePath: string; sha: string }>,
+    userId: SessionDataProps["isomerUserId"],
+    branchName: string
+  ): ResultAsync<string, GitFileSystemError | NotFoundError> {
+    let oldStateSha = ""
+    const efsVolPath = this.getEfsVolPathFromBranch(branchName)
+    const isStaging = this.isStagingFromBranchName(branchName)
+
+    return this.getLatestCommitOfBranch(repoName, branchName)
+      .andThen((latestCommit) => {
+        if (!latestCommit.sha) {
+          return errAsync(
+            new GitFileSystemError(
+              `Unable to find latest commit of repo: ${repoName} on branch "${branchName}"`
+            )
+          )
+        }
+        oldStateSha = latestCommit.sha as string
+        return okAsync(true)
+      })
+      .andThen(() =>
+        ResultAsync.combine(
+          items.map(({ filePath, sha }) =>
+            this.getFilePathStats(repoName, filePath, isStaging).andThen(
+              (stats) =>
+                okAsync({
+                  filePath,
+                  sha,
+                  stats,
+                })
+            )
+          )
+        )
+      )
+      .andThen((itemsWithStats) => {
+        const isFileConflict = !itemsWithStats.every(async (itemStats) => {
+          if (itemStats.stats.isDirectory()) {
+            // If it's a directory, skip the blob hash verification
+            return true
+          }
+
+          const result = await this.getGitBlobHash(
+            repoName,
+            itemStats.filePath,
+            isStaging
+          ).andThen((sha) => {
+            if (sha !== itemStats.sha) {
+              return okAsync(false)
+            }
+
+            return okAsync(true)
+          })
+
+          return result.isOk() && result.value
```

Comment:
Hmm yes, the `result.value` is the one that does the check for the unwrapped `ok`. Effectively there are 3 cases:
1. If an error occurred, `result.isOk()` is false, so the entire return is false.
2. If no error occurred but there is a sha mismatch, then result.value is false, so true && false becomes false.
3. If no error occurred and everything is good, then result.value is true, so true && true becomes true, which is what we want.

This is only specific for a single item, so it will loop through all the items before we determine the value of `isFileConflict`, which will then determine if we continue the function execution or not.